app/controllers/items_controller.rb

```diff
@@ -0,0 +1,28 @@
+require 'item_marketplace/presenters/items_index_presenter'
+require 'item_marketplace/presenters/new_item_presenter'
+
+class ItemsController < ApplicationController
+  def index
+    items = RepositoryContainer.get_repo(:item).all
+    @presenter = ItemsIndexPresenter.new(items)
+  end
+
+  def new
+    item = RepositoryContainer.get_repo(:item)
+    @presenter = NewItemPresenter.new(item)
+  end
+
+  def create
+    repo = RepositoryContainer.get_repo(:item)
+    item = repo.new_item(item_params)
+    if repo.save(item)
+      redirect_to :root
```

Comment:
A repository should not be returning a boolean wether it succeeded or not. A repository is just the boundary between the db and the app. This is a pattern often seen in ActiveRecord which has a ton of responsibilities. A repository either succeeds or translates or a db error